src/arrayMethodJoin.js

```diff
@@ -5,7 +5,36 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    function inputTest(input) {
+      if (input === null || input === undefined) {
+        return '';
+      } else {
+        return input;
+      }
+    };
+
+    function separatorTest(inputSeparator) {
+      if (inputSeparator === undefined) {
+        return ',';
+      } else if (inputSeparator === null) {
+        return 'null';
+      } else {
+        return inputSeparator;
+      }
+    };
+
+    for (let i = 0; i < this.length; i++) {
+      if (i < this.length - 1) {
+        result += inputTest(this[i]) + separatorTest(separator).toString();
```

Comment:
```suggestion
        result += inputTest(this[i]) + separatorTest(separator);
```
If you add an element to a string, it becomes string